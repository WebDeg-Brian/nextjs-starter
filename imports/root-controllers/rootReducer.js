import {
  UPDATE_VIEWPORT_DIMENSIONS,
  GET_BROWSER_INFO,
  GET_ENGINE_INFO,
  GET_MOBILE_INFO,
  GET_OS_INFO,
} from './rootTypes';
import { combineReducers } from 'redux';

const appState = {
  viewportWidth: null,
  viewportHeight: null,
  browserName: null,
  browserVersion: null,
  engineName: null,
  engineVersion: null,
  mobileVendor: null,
  mobileModel: null,
  OSName: null,
  OSVersion: null,
};

const app = (state = appState, action) => {
  switch (action.type) {
    case UPDATE_VIEWPORT_DIMENSIONS:
      return {
        ...state,
        viewportWidth: action.newWidth,
        viewportHeight: action.newHeight,
      };
    case GET_BROWSER_INFO:
      return {
        ...state,
        browserName: action.browserName,
        browserVersion: action.browserVersion,
      };
    case GET_ENGINE_INFO:
      return {
        ...state,
        engineName: action.engineName,
        engineVersion: action.engineVersion,
      };
    case GET_MOBILE_INFO:
      return {
        ...state,
        mobileVendor: action.mobileVendor,
        mobileModel: action.mobileModel,
      };
    case GET_OS_INFO:
      return {
        ...state,
        OSName: action.OSName,
        OSVersion: action.OSVersion,
      };
    default:
      return state;
  }
};

export default combineReducers({
  app,
});
