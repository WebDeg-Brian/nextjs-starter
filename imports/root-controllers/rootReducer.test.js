import rootReducer from './rootReducer';
import {
  UPDATE_VIEWPORT_DIMENSIONS,
  GET_BROWSER_INFO,
  GET_ENGINE_INFO,
  GET_MOBILE_INFO,
  GET_OS_INFO,
} from './rootTypes';
import createReducerTest from '../../tests/utils/createReducerTest';

const reducerTest = createReducerTest(rootReducer);

describe('Check if the state is mutated correctly', () => {
  reducerTest('Initial state', { type: 'UNRECOGNISED_TYPE' });
  reducerTest(UPDATE_VIEWPORT_DIMENSIONS, {
    type: UPDATE_VIEWPORT_DIMENSIONS,
    newWidth: 1024,
    newHeight: 512,
  });
  reducerTest(GET_BROWSER_INFO, {
    type: GET_BROWSER_INFO,
    browserName: 'Firefox',
    browserVersion: '63.0',
    fullBrowserVersion: '63.0.1',
  });
  reducerTest(GET_ENGINE_INFO, {
    type: GET_ENGINE_INFO,
    engineName: 'Gecko',
    engineVersion: '1.8.1',
  });
  reducerTest(GET_MOBILE_INFO, {
    type: GET_MOBILE_INFO,
    mobileVendor: 'Apple',
    mobileModel: 'iPhone 6s Plus',
  });
  reducerTest(GET_OS_INFO, {
    type: GET_OS_INFO,
    OSName: 'Windows',
    OSVersion: '8.1',
  });
});
