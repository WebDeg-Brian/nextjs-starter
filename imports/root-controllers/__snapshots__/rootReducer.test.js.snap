// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the state is mutated correctly GET_BROWSER_INFO 1`] = `
Object {
  "OSName": null,
  "OSVersion": null,
  "browserName": "Firefox",
  "browserVersion": "63.0",
  "engineName": null,
  "engineVersion": null,
  "fullBrowserVersion": "63.0.1",
  "mobileModel": null,
  "mobileVendor": null,
  "viewportHeight": null,
  "viewportWidth": null,
}
`;

exports[`Check if the state is mutated correctly GET_ENGINE_INFO 1`] = `
Object {
  "OSName": null,
  "OSVersion": null,
  "browserName": null,
  "browserVersion": null,
  "engineName": "Gecko",
  "engineVersion": "1.8.1",
  "fullBrowserVersion": null,
  "mobileModel": null,
  "mobileVendor": null,
  "viewportHeight": null,
  "viewportWidth": null,
}
`;

exports[`Check if the state is mutated correctly GET_MOBILE_INFO 1`] = `
Object {
  "OSName": null,
  "OSVersion": null,
  "browserName": null,
  "browserVersion": null,
  "engineName": null,
  "engineVersion": null,
  "fullBrowserVersion": null,
  "mobileModel": "iPhone 6s Plus",
  "mobileVendor": "Apple",
  "viewportHeight": null,
  "viewportWidth": null,
}
`;

exports[`Check if the state is mutated correctly GET_OS_INFO 1`] = `
Object {
  "OSName": "Windows",
  "OSVersion": "8.1",
  "browserName": null,
  "browserVersion": null,
  "engineName": null,
  "engineVersion": null,
  "fullBrowserVersion": null,
  "mobileModel": null,
  "mobileVendor": null,
  "viewportHeight": null,
  "viewportWidth": null,
}
`;

exports[`Check if the state is mutated correctly Initial state 1`] = `
Object {
  "OSName": null,
  "OSVersion": null,
  "browserName": null,
  "browserVersion": null,
  "engineName": null,
  "engineVersion": null,
  "fullBrowserVersion": null,
  "mobileModel": null,
  "mobileVendor": null,
  "viewportHeight": null,
  "viewportWidth": null,
}
`;

exports[`Check if the state is mutated correctly UPDATE_VIEWPORT_DIMENSIONS 1`] = `
Object {
  "OSName": null,
  "OSVersion": null,
  "browserName": null,
  "browserVersion": null,
  "engineName": null,
  "engineVersion": null,
  "fullBrowserVersion": null,
  "mobileModel": null,
  "mobileVendor": null,
  "viewportHeight": 512,
  "viewportWidth": 1024,
}
`;
